pkg/serviceaccount/legacy.go

```diff
@@ -126,6 +140,30 @@ func (v *legacyValidator) Validate(ctx context.Context, tokenData string, public
 			klog.V(4).Infof("Service account UID no longer matches %s/%s: %q != %q", namespace, serviceAccountName, string(serviceAccount.UID), serviceAccountUID)
 			return nil, fmt.Errorf("ServiceAccount UID (%s) does not match claim (%s)", serviceAccount.UID, serviceAccountUID)
 		}
+
+		if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.LegacyServiceAccountTokenTracking) {
+			for _, ref := range serviceAccount.Secrets {
+				if ref.UID == secret.UID {
+					warning.AddWarning(ctx, "", "Use tokens from the TokenRequest API or manually created secret-based tokens instead of auto-generated secret-based tokens.")
+					break
+				}
+			}
+			now := time.Now().UTC()
+			yesterday := now.AddDate(0, 0, -1).Format("2006-01-02")
+			today := now.Format("2006-01-02")
+			lastUsed := secret.GetAnnotations()[LastUsedAnnotationKey]
+			inUse := secret.GetLabels()[InUseLabelKey]
+			if (lastUsed != today && lastUsed != yesterday) || (inUse != today && inUse != yesterday) {
+				patchContent, err := json.Marshal(applyv1.ObjectMeta().WithLabels(map[string]string{LastUsedAnnotationKey: today}).WithAnnotations(map[string]string{InUseLabelKey: today}))
```

Comment:
this is rooting the patch at metadata, you need to use the ConfigMap apply configuration to patch a configmap

the generated patch should look like `{..., "metadata":{"annotations":{...}}}`

Reply:
ah. used the Secret apply configuration.

Reply:
yeah, sorry, Secret